### PR TITLE
chore(core): moves clearAllFiles into Entity class

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

**Comment by jdalsem** [comment]
sorry about all the white lines

File: docs/guides/upgrading.rst [hunk 1]
```diff
@@ -93,6 +93,7 @@ All the functions in ``engine/lib/deprecated-1.9.php`` were removed. See https:/
 All the functions in ``engine/lib/deprecated-1.10.php`` were removed. See https://github.com/Elgg/Elgg/blob/2.0/engine/lib/deprecated-1.10.php for these functions. Each ``@deprecated`` declaration includes instructions on what to use instead.
 
  * ``_elgg_manage_pagesetup``
+ * ``_elgg_clear_entity_files``: Use ``ElggEntity->clearAllFiles()``
```
Comment (hypeJunction) [inline docs/guides/upgrading.rst]: It was private, do we need it in the docs?

File: engine/classes/ElggEntity.php [hunk 2]
```diff
@@ -1837,12 +1837,25 @@ public function delete($recursive = true) {
...
 	}
 
 	/**
+	 * Removes all entity files in the dataroot
+	 *
+	 * @warning This only deletes the physical files and not their entities.
+	 *
+	 * @return bool
+	 */
+	public function clearAllFiles() {
```
Comment (jeabakker) [inline engine/classes/ElggEntity.php]: `protected` not public

File: docs/guides/upgrading.rst (hunk 1 above)
Comment (jeabakker) [reply docs/guides/upgrading.rst]: i suggest we make the entity function `protected` than maybe only mention it (no alternative)

**Review by jeabakker — DISMISSED** [review]

**Pull request merged**